lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -117,7 +117,17 @@ const startLighthouse = `
 
   // Give the main target model a moment to be available.
   // Otherwise, 'SDK.TargetManager.TargetManager.instance().mainTarget()' is null.
-  await new Promise(resolve => setTimeout(resolve, 1000));
+  const targetManager =
+    SDK.targetManager || (SDK.TargetManager.TargetManager || SDK.TargetManager).instance();
+  if (targetManager.mainTarget() === null) {
+    if (targetManager.observeTargets) {
+      await new Promise(resolve => targetManager.observeTargets({targetAdded: resolve}));
```

Comment:
Just to be on the safe side, could you add a blank `targetRemoved` listener

Reply:
would it error without?

Reply:
`targetRemoved` is expected here, would throw when it tries to run an unset property:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/core/sdk/TargetManager.ts;l=215?q=TargetManager&ss=chromium%2Fchromium%2Fsrc:third_party%2Fdevtools-frontend%2Fsrc%2F